### PR TITLE
GEOMESA-891 Allow geoserver requests to specify the query strategy

### PR DESCRIPTION
**Pull request by elahrvivaz**
* Append e.g. '&viewparams=strategy:attribute' to the request
* Valid strategies are currently Z3, ST, RECORD, ATTRIBUTE
* Note: not all queries will work with a given strategy

Signed-off-by: Emilio Lahr-Vivaz <elahrvivaz@ccri.com>

### CODE REVIEW

File: geomesa-accumulo/geomesa-accumulo-datastore/src/main/scala/org/locationtech/geomesa/accumulo/index/QueryPlanner.scala [hunk 1]
```diff
@@ -222,6 +226,38 @@ object QueryPlanner {
...
     QueryPlanner.setReturnSft(query, sft)
+    // handle any params passed in through geoserver
+    QueryPlanner.handleGeoServerParams(query)
+  }
+
+  /**
+   * Checks the 'view params' passed in through geoserver and converts them to the appropriate query hints.
+   * This kind of a hack, but it's the only way geoserver exposes custom data to the underlying data store.
+   *
+   * Note - keys in the map are always uppercase.
```
Comment (cne1x) [inline geomesa-accumulo/geomesa-accumulo-datastore/src/main/scala/org/locationtech/geomesa/accumulo/index/QueryPlanner.scala]: These comments are very useful; thanks!

**Comment by cne1x** [comment]
@elahrvivaz There's a conflict between this PR and the one yesterday to handle the namespace/nulls.  Do you mind rebasing on master to address that?

**Comment by elahrvivaz** [comment]
@cne1x Fixed conflicts, thanks.

**Comment by cne1x** [comment]
(Finally) merged as commit aba7a46e49629c744b17eb7f39c08859ee7c34e8

**Pull request closed without merging**